### ISSUE TITLE
Dialog is not WCAG 2.1 compliant for reflow checkpoint

### ISSUE THREAD

**Issue by ranaJPMChase** [jpmorganchase/salt-ds#4697]
### Latest version

- [x] I have tested the latest version

### Description

Content should be seen properly! When screen is 320 by 256 pixels. Following WCAG rules (DAKB)

### Steps to reproduce

In chrome open [url.](https://storybook.saltdesignsystem.com/?path=/story/core-dialog--long-content)
Click on long content dialog. In left nav. 
Right click on screen then select inspect. Select "toggle device toolbar" icon to make screen responsive.
Select dimensions responsive and 320 pixels width and 256 pixels height
Check if you can see the actual dialog content.

### Expected behavior

user should be able to see dialog content properly. Right now only a tiny line shows up through which user can see the content. 

### Package name(s)

Core (@salt-ds/core)

### Package version(s)

did testing on https://storybook.saltdesignsystem.com/?path=/story/core-dialog--long-content

### Browser

- [x] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Microsoft Edge

### Operating system

- [ ] macOS
- [x] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### Are you a JPMorgan Chase & Co. employee?

- [x] I am an employee of JPMorgan Chase & Co.

**Comment by origami-z** [comment]
@jake-costa could you please advice on 256px height problem? This could mean we need a different header pattern for ultra small screen where the whole dialog should scroll. cc @navkaur76 @bhoppers2008

**Comment by joshwooding** [comment]
> [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/jake-costa?size=32">@jake-costa](https://github.com/jake-costa?rgh-link-date=2025-02-17T11%3A17%3A30.000Z) could you please advice on 256px height problem? This could mean we need a different header pattern for ultra small screen where the whole dialog should scroll. cc [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/navkaur76?size=32">@navkaur76](https://github.com/navkaur76?rgh-link-date=2025-02-17T11%3A17%3A30.000Z) [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/bhoppers2008?size=32">@bhoppers2008](https://github.com/bhoppers2008?rgh-link-date=2025-02-17T11%3A17%3A30.000Z)

I caught up with @jake-costa on this, he said it's a problem that will need to be fixed. It seem the easiest fix it to change the height of the dialog at smaller viewport sizes which should be tackled in #3856

**Comment by ranaJPMChase** [comment]
@joshwooding if header text is little longer lets say more than 3 words then in 320 by 256 buttons are not accessible as well and user cant scroll.

**Comment by mark-tate** [comment]
Clove Goal: complete by EOS

**Comment by origami-z** [comment]
May 1 - Will have a look next week

**Comment by mark-tate** [comment]
Will look when we have capacity

**Comment by mark-tate** [comment]
AllSpice Goal: meeting to understand what is outstanding, delivery date to be defined

**Comment by joshwooding** [comment]
Docs/examples will be updated, list of changes to be added in a new ticket @navkaur76 . @bhoppers2008 to review

**Comment by mark-tate** [comment]
Garlic Goal: @jake-costa & @navkaur76 to VQA PR

**Comment by jake-costa** [comment]
@navkaur76 let me know if we need to connect, otherwise can you provide an update on next steps?
